### DISCUSSION TITLE
No response from request api(longer than 30min) while using SimpleHttpOperator

### DISCUSSION THREAD

**Discussion by yuanke7** [apache/airflow#18321] [General]
### Apache Airflow version

2.1.2

### Operating System

CentOS Linux 7

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==2.0.0
apache-airflow-providers-celery==2.0.0
apache-airflow-providers-cncf-kubernetes==2.0.0
apache-airflow-providers-docker==2.0.0
apache-airflow-providers-elasticsearch==2.0.2
apache-airflow-providers-ftp==2.0.0
apache-airflow-providers-google==4.0.0
apache-airflow-providers-grpc==2.0.0
apache-airflow-providers-hashicorp==2.0.0
apache-airflow-providers-http==2.0.0
apache-airflow-providers-imap==2.0.0
apache-airflow-providers-microsoft-azure==3.0.0
apache-airflow-providers-mysql==2.0.0
apache-airflow-providers-postgres==2.0.0
apache-airflow-providers-redis==2.0.0
apache-airflow-providers-sendgrid==2.0.0
apache-airflow-providers-sftp==2.0.0
apache-airflow-providers-slack==4.0.0
apache-airflow-providers-sqlite==2.0.0
apache-airflow-providers-ssh==2.0.0

### Deployment

Docker-Compose

### What happened

I'm using SimpleHttpOperator to request an api which get reponse longer than 30min often, but task is still in running state even api has already returned a response. 
Here is one of my task code:
`
psd_report = SimpleHttpOperator(
    task_id='psd_report',
    method='GET',
    http_conn_id=WIN_SERVER_1,
    endpoint='general/psd_report',
    response_check=lambda response : response.json().get('Success'),
    response_filter=lambda response : response.json(),
    execution_timeout=timedelta(hours=1),
    dag=dag,
)
`
Here is my api response data:
`
{"success":True}
`

### What you expected to happen

I expected that task using SimpleHttpOperator could switch state from running to success so that downstream task could get into the queue.

### How to reproduce

Creat a **SimpleHttpOperator** task, request an api that need at least 30 min or more to response. Trigger this task  you could find even api has returned a response, the task is still in running state.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
You certainly need to provide more information and more details, what exactly you do, some logs and results of your investigation - what did you try and what did not work before you can make it into an "airflow issue". I converted it into discussion, and I think you need to provide more information for anyone to be able to even try to help you with your problem.

**Reply by yuanke7** [reply]
Thanks, potiuk I will put my code on it.

**Reply by yuanke7** [reply]
Pls help...

**Reply by potiuk** [reply]
It is likely your API/Firewall/communication problem. I am not sure anyone can help you with that - there is nothing in Airflow that has 30 minutes cut-off on the request. 

The best I can advice is that you try to replicate it with pure python - copying the code the HTTP operator use, experiment with it and work with your deployment specialist to figure out where the problem is. When you connect to airflow worker (again depending on your deployment) you should be able to write a simple python script that does exactly what Simple HTTP operator does - and experiment: https://github.com/apache/airflow/blob/main/airflow/providers/http/hooks/http.py#L129
